src/mscorlib/shared/System/IO/FileStream.Unix.cs

```diff
@@ -189,25 +186,21 @@ private static Interop.Sys.OpenFlags PreOpenConfigurationFromOptions(FileMode mo
         }
 
         /// <summary>Gets a value indicating whether the current stream supports seeking.</summary>
-        public override bool CanSeek => CanSeekCore;
+        public override bool CanSeek => CanSeekCore(_fileHandle);
 
         /// <summary>Gets a value indicating whether the current stream supports seeking.</summary>
         /// <remarks>Separated out of CanSeek to enable making non-virtual call to this logic.</remarks>
-        private bool CanSeekCore
+        private bool CanSeekCore(SafeFileHandle fileHandle)
```

Comment:
It'd be good to add a comment about why this is being passed in, so that someone in the future doesn't "simplify" it back